fairlearn/post_processing/roc_curve_based_post_processing.py

```diff
@@ -147,18 +154,27 @@ def _validate_estimator(self):
                 not callable(fit_function):
             raise ValueError(MISSING_FIT_PREDICT_ERROR_MESSAGE)
 
-    def _validate_input_data(self, X, protected_attribute, y=None):
-        if type(X) != type(protected_attribute) or (y is not None and type(X) != type(y)) or \
-                type(X) not in [list, np.ndarray]:
-            raise ValueError(INPUT_DATA_CONSISTENCY_ERROR_MESSAGE
-                             .format(type(X).__name__, type(y).__name__,
-                                     type(protected_attribute).__name__))
-
-        if len(X) == 0 or len(protected_attribute) == 0 or (y is not None and len(y) == 0):
+    def _validate_input_data(self, X, aux_data, y=None):
+        allowed_input_types = [list, np.ndarray, pd.DataFrame, pd.Series]
+        if type(X) not in allowed_input_types or \
+                type(aux_data) not in allowed_input_types or \
+                (y is not None and type(y) not in allowed_input_types):
+            raise ValueError(INPUT_DATA_FORMAT_ERROR_MESSAGE
+                             .format(type(X).__name__,
+                                     type(y).__name__,
+                                     type(aux_data).__name__))
+
+        if len(X) == 0 or len(aux_data) == 0 or (y is not None and len(y) == 0):
             raise ValueError(EMPTY_INPUT_ERROR_MESSAGE)
 
-        if len(X) != len(protected_attribute) or (y is not None and len(X) != len(y)):
-            raise ValueError(DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE)
+        if y is None:
+            if len(X) != len(aux_data) or (y is not None and len(X) != len(y)):
+                raise ValueError(DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE
+                                 .format("X and aux_data"))
+        else:
+            if len(X) != len(aux_data) or (y is not None and len(X) != len(y)):
+                raise ValueError(DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE
```

Comment:
We will need to sit down and review all the exceptions we raise, so that we use the same exception type for the same sort of errors

Reply:
I only use ValueError when it's an error from an input, for example bad format/length, etc.; NotImplementedError when I don't want to implement something, NotFittedException when a predictor isn't fitted (and therefore not a predictor). That's consistent throughout

Reply:
So no RuntimeErrors?

Reply:
What would I use those for?

Reply:
It's what I happened on for some cases where you've gone for ValueError. I'm not arguing for one or the other, but we will need to sit down and standardise.

Reply:
Sure, we can do that.

Reply:
After this PR, though :-)